agent/api/task/task.go

```diff
@@ -360,6 +360,8 @@ func (task *Task) PostUnmarshalTask(cfg *config.Config,
 
 	task.initializeContainersV3MetadataEndpoint(utils.NewDynamicUUIDProvider())
 	task.initializeContainersV4MetadataEndpoint(utils.NewDynamicUUIDProvider())
+	task.initializeContainersGQLMetadataEndpoint(utils.NewDynamicUUIDProvider())
```

Comment:
nit: can we refactor this a little and create only one DynamicUUIDProvider in line 360 and then pass it to the different `initializeContainers` functions?